server-engine: fix crypto-length in Create

viktorTarasov opened the pull request:
In request paylod to create symmetric key the
length of cryptographic algorithm is optional.

As Peter suggested, in this case the strongest allowed key size
is used.

PeterHamilton commented on kmip/services/server/crypto/engine.py:
```diff
@@ -86,7 +86,13 @@ def create_symmetric_key(self, algorithm, length):
+            #   the strongest allowed key size is used.
+            sizes = list(cryptography_algorithm.key_sizes)
+            sizes.sort()
+            length = sizes[len(sizes) - 1]
```
To get the last item, you can just do:
    length = sizes[-1]

PeterHamilton commented on kmip/services/server/engine.py:
```diff
@@ -671,16 +671,8 @@ def _process_create(self, payload):
-            # TODO (peterhamilton) The cryptographic length is technically not
-            # required per the spec. Update the CryptographyEngine to accept a
-            # None length, allowing it to pick the length dynamically. Default
-            # to the strongest key size allowed for the algorithm type.
```
Now that I think about it, the cryptography engine's `create_symmetric_key` method will need to also include the `key_length` in the result dict, since the length is needed for storing the whole managed object (line 699 below).

PeterHamilton commented on kmip/tests/unit/services/server/test_engine.py:
```diff
@@ -1607,51 +1607,6 @@ def test_create_omitting_attributes(self):
-            "Processing operation: Create"
-        )
-        e._logger.reset_mock()
-
```
This whole block needs to be replaced with a test that verifies that the strongest key size is picked when the key length is not specified.

viktorTarasov commented:
Sorry, I have not seen your comments and rebased the branch.
In this rebase I added crypto-length into response payload, if it was not present in request payload.

I will review the code, with your other comments in head, tomorrow ...
 ... and fix coveralls missings.

viktorTarasov commented:
replaced by #170
